cf-custom-resources/lib/dns-cert-validator.js

```diff
@@ -496,6 +491,38 @@ const updateRecords = function (
     .promise();
 };
 
+// getAllAliases gets all aliases out from a string. For example:
+// {"frontend": ["test.foobar.com", "foobar.com"], "api": ["api.foobar.com"]} will return
+// ["test.foobar.com", "foobar.com", "api.foobar.com"].
+const getAllAliases = function (aliases) {
+  let obj;
+  try {
+    obj = JSON.parse(aliases || "{}");
+  } catch (error) {
+    throw new Error(`Cannot parse ${aliases} into JSON format.`);
+  }
+  var aliasList = [];
+  for (var m in obj) {
+    aliasList.push(...obj[m]);
+  }
+  return new Set(aliasList.filter(function (itm) {
+    return getDomainType(itm) != domainTypes.OtherDomainZone;
+  }));
+};
+
+const getDomainType = function (alias) {
+  switch (true) {
+    case domainTypes.EnvDomainZone.regex.test(alias):
```

Comment:
nit: can we just do:
```suggestion
  if (domainTypes.EnvDomainZone.regex.test(alias)) {
    return return domainTypes.EnvDomainZone;
  }
```